testing_and_setup/compass/utility_scripts/LIGHTparticles/make_particle_file.py

```diff
@@ -0,0 +1,490 @@
+#!/usr/bin/env python
+"""
+File writes a particle input dataset for use in MPAS-O / E3SM.
+
+Example usage is
+
+    python make_particle_file.py -i init.nc -g graph.info.part.6 \
+            -o particles.nc -p 6 --spatialfilter SouthernOceanXYZ \
+            --downsample=1
+
+Phillip J. Wolfram
+Last Modified: 08/03/2018
+"""
+
+import netCDF4
+import numpy as np
+from scipy import spatial
+from scipy import sparse
+from pyamg.classical import split, interpolate as amginterp
```

Comment:
@pwolfram, line 19 should be wrapped in a try block per our call with ANL and @mark-petersen's error. This of course will break if the user tries the `--downsample` flag without `pyamg`, but that should be fine. For whatever reason, `pyamg` is not an easy install (took many tries for me).

Reply:
@bradyrx thanks for the suggestion.  I agree, put it in a try block, and have it actually fail only when the library is needed.  That will keep our nightly testing suite more accessible and portable across users and machines.

Reply:
Ideally we eventually need a better solution than `pyamg` for coarsening, but I don't think it's crucial for this PR. It's a feasible solution for oQU240, but takes ~40min for 60to30 and ~20hr for 30to10.

Reply:
Note my current hacky solution here: https://gist.github.com/bradyrx/ddb6c944e425ac5e63b8e8e7361809dd

I'm just running `make_particle_file.py` without downsampling, and then removing full columns with that gist by going through lat/lon with a fixed stride.

This approximately pulls columns at random due to the unstructured mesh indexing, but I don't suggest it as a formal method in ocean/develop.

Here, white dots are seeding at every grid cell and red are after running my script:

![](https://i.imgur.com/o6A3uq4.jpg)